### ISSUE TITLE
Onyxia 40 Bugs

### ISSUE THREAD

**Issue by Day36512** [ZhengPeiRu21/mod-individual-progression#91]
Hey there!

Just want to say that I looove your module.

Having a couple issues with ony. First few times upon entering, the raid kept reseting to 10 man even though it needed to be 10 H. Recompiled, restarted, set my character back to stage 1, entered with a second character outside who set it to 10H. Finally got the correct instance. In other words, if the leader enters first without anyone else, the raid gets set to 10 man normal. Tried testing again after instance properly set to 10H and didn't reset to N again (Did this all without addons).

But now rather than whelps spawning in caves and aggroing to characters in the middle upon air phase, Ony goes up into the air and whelps drop in line from under her. They also don't engage with players and are "too far away" even though they're visually on the ground not moving. Tested the whelp eggs in the caves and those whelps work fine when you runover eggs, but it seems like the spawned whelps aren't doing anything and just get stuck in the air but drop to the ground visually, as i can set gm fly on and aggro them down.Honeslty they can have a 10000 yard agro radius and it wouldn't hurt anything, but it's not blizzlike to have them spawn in a straight line along her initial fly path. It seems like spell 900003 is not setting a spawn position for them except under the boss. 

Final note: Ony is supposed to have a much larger aggro radius, about 40-50 yards so that she aggros when players enter the room. Doesn't look like Azerothcore master has it right either though.

Let me know if you can't duplicate the bug! Would be curious to see if something is just going terribly wrong on my end.

**Comment by ZhengPeiRu21** [comment]
Thank you for the useful and detailed report! I'll see what I can do to update the module and AzerothCore master when necessary.

**Comment by Day36512** [comment]
Just wanted to share the change I made with lua to fix the whelp spawn fix. You may be able to convert it to cpp fairly easily.
[Onyxia.zip](https://github.com/ZhengPeiRu21/mod-individual-progression/files/11908466/Onyxia.zip)

**Comment by ZhengPeiRu21** [comment]
> Just wanted to share the change I made with lua to fix the whelp spawns. You may be able to convert it to cpp fairly easily. [Onyxia.zip](https://github.com/ZhengPeiRu21/mod-individual-progression/files/11908466/Onyxia.zip)

Thank you for sharing! I will try and get this bug fixed soon.

**Comment by Grimfeather** [comment]
I've encountered an issie with Onyxia's lair as well.
I run Individual-Progression together with NPCbots and tried to enter the lvl60 40man version of Onyxia's lair with 19 bots 
https://i.gyazo.com/86e86b0c21741931dcec5f929c9e0500.jpg

Only 9 of them entered together with me. Into the 10 man version.
It had lvl 60-63 Onyxian lair guards and whelps as well during stage 2
Onyxia herself could have been the level 80 version, not sure. 1623k health I think she had

Wasn't able to kill her. When I tried to run back inside there was another problem,
 as a ghost I was not able to get back in to resurrect.
